rct229/rulesets/ashrae9012019/section19/section19rule21.py

```diff
@@ -0,0 +1,498 @@
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0]",
+            data_items={"climate_zone": ("baseline", "weather/climate_zone")},
+        )
+
+    class RMDRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section19Rule21.RMDRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                each_rule=Section19Rule21.RMDRule.HVACRule(),
+                index_rmr="baseline",
+                list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            )
+
+        def create_data(self, context, data):
+            rmd_b = context.baseline
+            rmd_p = context.proposed
+
+            dict_of_zones_and_terminal_units_served_by_hvac_sys_b = (
+                get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmd_b)
+            )
+
+            baseline_hvac_system_dict = get_baseline_system_types(rmd_b)
+
+            zone_data = {}
+            hvac_systems_and_assoc_zones_largest_exhaust_source = {}
+            for hvac_id_b in find_all(
+                "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].id",
+                rmd_b,
+            ):
+                zone_data[hvac_id_b] = {
+                    "serves_zones_with_systems_likely_exhausting_toxic_etc": False,
+                    "serves_kitchen_space": False,
+                    "all_lighting_space_types_defined": True,
+                    "all_ventilation_space_types_defined": False,
+                }
+
+                for zone_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b[
+                    hvac_id_b
+                ]["zone_list"]:
+                    space_p = find_one(
+                        f'$.buildings[*].building_segments[*].zones[*][?(@.id = "{zone_id_b}")].spaces[*]',
+                        rmd_p,
+                    )
+
+                    if space_p is not None and space_p.get(
+                        "lighting_space_type", False
+                    ):
+                        lighting_space_type_p = space_p["lighting_space_type"]
+                        if (
+                            lighting_space_type_p
+                            == LIGHTING_SPACE.LABORATORY_EXCEPT_IN_OR_AS_A_CLASSROOM
+                        ):
+                            zone_data[hvac_id_b][
+                                "serves_zones_with_systems_likely_exhausting_toxic_etc"
+                            ] = True
+
+                        if (
+                            lighting_space_type_p
+                            == LIGHTING_SPACE.FOOD_PREPARATION_AREA
+                        ):
+                            zone_data[hvac_id_b]["serves_kitchen_space"] = True
+
+                    else:
+                        zone_data[hvac_id_b]["all_lighting_space_types_defined"] = False
```

Comment:
Will need to rewrite this logic - note that it is not just one space, it is space list inside a zone.
```python
# any logic
space_lighting_space_type_list = [space.get("lighting_space_type", None) for space in spaces_p]
serves_zones_with_systems_likely_exhausting_toxic_etc = LIGHTING_SPACE.LABORATORY_EXCEPT_IN_OR_AS_A_CLASSROOM in space_lighting_space_type_list
serves_kitchen_space = LIGHTING_SPACE.FOOD_PREPARATION_AREA in space_lighting_space_type_list
all_lighting_space_types_defined = all(space_lighting_space_type_list)
```

Reply:
Addressed.